### DISCUSSION TITLE
Combining Predictions and Uncertainty

### DISCUSSION THREAD

**Discussion by FisheryScience89** [sdmTMB/sdmTMB#313] [Q&A] [answered]
If one had to combine predictions from two models in some way, would using nsim on both models and then combining the result be a valid way to get at uncertainty in the combined prediction?

For example, combined prediction = model a prediction * model b prediction?

**Comment by seananderson** [comment, marked as answer]
Yes, you absolutely could do that. We did that in this recent paper: https://doi.org/10.1002/pan3.10554

However, a couple caveats. If it's something that can be crammed into one of the existing delta models then that can be done within the model. If we're talking about uncertainty on predictions and space, the fastest way right now is probably to simulate from the model anyways and summarize that (we might have a faster way to do that soon). The most robust way to do this would be with MCMC sampling (rather than by using the `nsim` argument to take draws from the parameters assuming an MVN distribution), although that can be slow for big models (and if all you are trying to do is visualize spatial uncertainty then approximate samples might be good enough). The tmbstan sampling can often be sped up somewhat by fixing the log decorrelation rate (`ln_kappa`) at the maximum likelihood estimate before sampling with tmbstan (plus using priors elsewhere). There is an example in the Bayesian vignette https://pbs-assess.github.io/sdmTMB/articles/web_only/bayesian.html

**Reply by FisheryScience89** [reply]
Great, thanks!
